rmw_connext_cpp/CMakeLists.txt

```diff
@@ -94,14 +94,33 @@ add_custom_command(
   VERBATIM
 )
 
+# determine major version of Connext
+find_file(connext_version_header "ndds_version.h"
+  PATHS ${Connext_INCLUDE_DIRS}
+  PATH_SUFFIXES "ndds"
+  NO_DEFAULT_PATH)
+if(NOT connext_version_header)
+  message(FATAL_ERROR "Failed to find 'ndds/ndds_version.h' in '${Connext_INCLUDE_DIRS}'")
+endif()
+file(STRINGS "${connext_version_header}" connext_define_major_version LIMIT_COUNT 1 REGEX "#define RTI_DDS_VERSION_MAJOR  [0-9]+")
+if("${connext_define_major_version}" STREQUAL "")
+  message(FATAL_ERROR "Failed to find '#define RTI_DDS_VERSION_MAJOR' in '${connext_version_header}'")
+endif()
+string(REGEX REPLACE ".* ([0-9]+)" "\\1" connext_major_version "${connext_define_major_version}")
+
 # patch the generate code for raw data
+if("${connext_major_version}" LESS 6)
+  set(patch_files_subdirectory "patch_files")
+else()
+  set(patch_files_subdirectory "patch_files_v6")
```

Comment:
@dirk-thomas meta: do we foresee this happening often? Should we come up with a naming scheme that scales better?

Reply:
Maybe v7 is going to break this again? Then I would just add a v7 directory.

Reply:
Yeah, and duplicate all most files in vN-1 in the process :/ Today it's just two though, I guess it's fine.

Reply:
The v6 patch files have differences compared to the v5 ones: aff4607937f3c135234e735a34825bc17468e75b